targets/TARGET_TOSHIBA/TARGET_TMPM066/sleep.c

```diff
@@ -0,0 +1,66 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sleep_api.h"
+
+// number of warm-up cycle = warm-up time to set / input frequency cycle (s)
+// number of 3*10^-6 (s) / (1/12 (MHz)) = 60000 = 0xea60
+#define CG_WUODR_INT     ((uint16_t)0xea60)
+
+void external_losc_enable(void);
+
+void hal_sleep(void)
+{
+    // Set low power consumption mode IDLE
+    CG_SetSTBYMode(CG_STBY_MODE_IDLE);
+    // Enter idle mode
+    __WFI();
+}
+
+void hal_deepsleep(void)
+{
+    // Set low power consumption mode STOP1
+    CG_SetSTBYMode(CG_STBY_MODE_STOP1);
+    // Setup warm up time
+    CG_SetWarmUpTime(CG_WARM_UP_SRC_OSC_EXT_HIGH, CG_WUODR_INT);
+    // Enter stop1 mode
+    __WFI();
+    // Switch over from IHOSC to EHOSC
+    external_losc_enable();
```

Comment:
Switching from Internal Oscillator to External Oscillator takes as WarmUpTime (CG_WUODR_INT).
600mW is the peak power consumption at T = 85degC.
When using IOSC 10MHz,<WUP[15:0]> = 0x2700
Warm-up time = (0x2700 + 0x10) × (1/10 MHz)
= 0x2710 × 100ns
= 10000 × 100ns
= 1.0 ms
When using EOSC 16MHz,<WUPT[15:0]> = 0x3E70
Warm-up time = (0x3E70 + 0x10) × (1/16 MHz)
= 0x3E80 × 62.5 ns
= 16000 × 62.5 ns
= 1.0 ms